# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/animation/KeyframeEffect.h
# Commit: 6f62586a139b
# Full Hash: 6f62586a139b1fcc4817dddd924400958d89a6bb
# Author: Zach Hoffman <zach@zrhoffman.net>
# Date: 2023-12-18 17:17:57
# Regressor Bug: 1846516
# File Overlap Count: 2
# Description:
#   Bug 1846516 - [css-properties-values-api] Animate custom properties in a discrete way. r=firefox-style-system-reviewers,emilio
#   
#   This patch also updates the bug ID for a FIXME leftover from bug 1840478
#   to bug 1869476, since the same FIXME is added in D190758.
#   
# ==============================================================================

diff -r 729abedf9bda -r 6f62586a139b dom/animation/KeyframeEffect.h
--- a/dom/animation/KeyframeEffect.h	Mon Dec 18 09:24:08 2023 +0000
+++ b/dom/animation/KeyframeEffect.h	Mon Dec 18 09:24:08 2023 +0000
@@ -333,7 +333,7 @@
     // We cannot use getters_AddRefs on StyleAnimationValue because it is
     // an incomplete type, so Get() doesn't work. Instead, use GetWeak, and
     // then assign the raw pointer to a RefPtr.
-    result.mServo = mBaseValues.GetWeak(aProperty.mID, &hasProperty);
+    result.mServo = mBaseValues.GetWeak(aProperty, &hasProperty);
     MOZ_ASSERT(hasProperty || result.IsNull());
     return result;
   }
@@ -455,7 +455,8 @@
   // least one animation value that is composited with the underlying value
   // (i.e. it uses the additive or accumulate composite mode).
   using BaseValuesHashmap =
-      nsRefPtrHashtable<nsUint32HashKey, StyleAnimationValue>;
+      nsRefPtrHashtable<nsGenericHashKey<AnimatedPropertyID>,
+                        StyleAnimationValue>;
   BaseValuesHashmap mBaseValues;
 
  private: